### ISSUE TITLE
New rule DL3062 is overly strict

### ISSUE THREAD

**Issue by tkmph** [hadolint/hadolint#1130]
We use go code (instead of, e.g., shell) to perform builds. We compile and execute the go code from the Dockerfile as follows:
```
RUN go run <path-to-go-file> <arguments>
```

The [new rule DL3062](https://github.com/hadolint/hadolint/pull/1111) flags this use:
```
<error column="1" line="14" message="Pin versions in go. Instead of `go install <package>` use `go install <package>@<version>`" severity="warning" source="DL3062"/>
```

This is overly strict. The go file is managed by version control and matches exactly the other code at the same version. I don't think that the rule should flag invocations of `go run` that execute specific go files (`go run /path/to/file.go`).
